src/app/address/controllers/addressConfirm.js

```diff
@@ -3,17 +3,38 @@ const BaseController = require("hmpo-form-wizard").Controller;
 class AddressConfirmController extends BaseController {
   locals(req, res, callback) {
     super.locals(req, res, (err, locals) => {
-      const buildingNumber = req.sessionModel.get("addressLine1");
-      const streetName = req.sessionModel.get("addressLine2");
-      const town = req.sessionModel.get("addressTown");
-      const postcode = req.sessionModel.get("addressPostcode");
+      const sessionAddresses = req.sessionModel.get("addresses");
+      const addresses = JSON.parse(sessionAddresses);
 
-      const formattedAddress = `${buildingNumber}<br>${streetName},</br>${town},</br>${postcode},</br>`;
+      const currentAddress = addresses.shift();
```

Comment:
In theory shouldnt be able to get to the confirmation page and have no addresses in the session.

Reply:
You can't get to the next page in the step without answering the previous questions, and there is a `prereq` check on every step to make sure the previous steps are completed as expected.

However, it's probably not a bad idea to do some defensive programming, just in case.